nemo_text_processing/g2p/modules.py

```diff
@@ -371,6 +379,8 @@ def _parse_as_cmu_dict(phoneme_dict_path, use_stresses=False, stress_symbols=[],
                         for stress_symbol in stress_symbols:
                             pronunciation = pronunciation.replace(stress_symbol, '')
                     symbols.update(pronunciation)  # This will insert each char individually
+                    if use_chars:
+                        symbols.update(word)
```

Comment:
Hmm, this may add some punctuation in the dictionary. It probably won't mess anything up, but I think it'll override the punctuation list set by the tokenizer, which may lead to some unexpected behavior (might mess with reserved symbols?).

Not sure what the best way of handling this is that's still generic across languages.

Reply:
Maybe we just strip all `string.punctuation` from the symbol set at the end of dictionary loading... But then that doesn't catch other languages' punctuation marks :T

Reply:
It would be safer to strip with the regex r"[a-zA-ZÀ-ÿ\d]".

Currently when we have contractions (eg. don't) and compound phrases (eg. long-term) do we process the words as-is or do we process them after stripping the punctuation?

Reply:
We may have to revisit this if/when we add non-Latin languages, depending on how they're going to be represented.

I'd say process words as-is in the dictionary (i.e. what the original behavior was) and only strip them out of the symbol set for the G2P class. Some of the punctuation may be valid and will be added back in by the tokenizer, e.g. apostrophes, so we want the dictionary lookup to still match.